samples/subsys/ipc/ipc_service/remote/src/main.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <drivers/ipm.h>
+#include <sys/printk.h>
+#include <device.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <ipc/ipc_service.h>
+
+#define APP_TASK_STACK_SIZE (1024)
+K_THREAD_STACK_DEFINE(thread_stack, APP_TASK_STACK_SIZE);
+static struct k_thread thread_data;
+
+static volatile unsigned int received_data;
+
+static K_SEM_DEFINE(data_rx_sem, 0, 1);
+static K_SEM_DEFINE(ept_bound_sem, 0, 1);
+
+static struct ipc_ept *ept;
+
+static int ept_bound(void *priv)
+{
+	k_sem_give(&ept_bound_sem);
+	return 0;
+}
+
+static int ept_recv(void *data, size_t len, void *priv)
+{
+	received_data = *((unsigned int *) data);
+	k_sem_give(&data_rx_sem);
+	return 0;
+}
+
+static int ept_error(const char *message, void *priv)
+{
+	printk("Endpoit error: %s", message);
+	return 0;
+}
+
+static unsigned int receive_message(void)
+{
+	k_sem_take(&data_rx_sem, K_FOREVER);
+	return received_data;
+}
+
+static int send_message(unsigned int message)
+{
+	return ipc_service_send(ept, &message, sizeof(message));
+}
+
+void app_task(void *arg1, void *arg2, void *arg3)
+{
+	ARG_UNUSED(arg1);
+	ARG_UNUSED(arg2);
+	ARG_UNUSED(arg3);
+	int status = 0;
+	unsigned int message = 0U;
+
+	printk("\r\nIPC Service [remote] demo started\r\n");
+
+	static struct ipc_ept_cfg ept_cfg = {
+		.name = "ep",
+		.prio = -1,
+		.priv = NULL,
+		.cb = {
+			.bound    = ept_bound,
+			.received = ept_recv,
+			.error    = ept_error,
+		},
+	};
+
+	status = ipc_service_register_endpoint(&ept, &ept_cfg);
```

Comment:
How about creating two endpoints to demonstrate flexibility of IPC Service API?

Reply:
A second endpoint has been added.